modules/aioseop_sitemap.php

```diff
@@ -4284,6 +4373,87 @@ public function get_all_post_priority_data( $include = 'any', $status = 'publish
 
 			$links = $this->get_prio_from_posts( $posts, $this->get_default_priority( 'post', true ), $this->get_default_frequency( 'post', true ) );
 			$links = array_merge( $links, $this->get_archive_prio_from_posts( $posts ) );
+
+			$is_sitemap_indexes_disabled = empty( $this->options['aiosp_sitemap_indexes'] );
+			if ( $is_sitemap_indexes_disabled || ( ! $is_sitemap_indexes_disabled && 'page' === $include ) ) {
+				$links = $this->get_prio_freq_static_homepage( $links );
+				$links = $this->update_woocommerce_shop_timestamp( $links );
+			}
+
+			return $links;
+		}
+
+		/**
+		 * The get_prio_freq_static_homepage() function.
+		 *
+		 * Sets the priority and frequency for the homepage if it is static.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param array $links
+		 * @return array $links
+		 */
+		private function get_prio_freq_static_homepage( $links ) {
+			if ( 0 === get_option( 'page_on_front' ) ) {
+				return $links;
+			}
+			$prio = $this->options['aiosp_sitemap_prio_homepage'];
+			$freq = $this->options['aiosp_sitemap_freq_homepage'];
+
+			$homepage_url = get_site_url() . '/';
+			$count = count( $links );
+			for ( $i = 0; $i < $count; $i++ ) {
```

Comment:
This For loop could be reduced with either...

```
$test_key = array_search( $homepage_url, array_column( $links, 'loc' ) );
$test_keys = array_keys( array_column( $links, 'loc' ), $homepage_url );
```

Reply:
Done.